### PR TITLE
Aliyun: Migrate tests to junit5 for aliyun client factory

### PR DESCRIPTION
**Pull request by DaVincii**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 #7160 ".

### CODE REVIEW

File: aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java [hunk 1]
```diff
@@ -22,43 +22,30 @@
...
-        AliyunClientFactories.defaultFactory());
+    assertThat(AliyunClientFactories.defaultFactory()).as("Default client should be singleton").isEqualTo(AliyunClientFactories.defaultFactory());
 
     AliyunClientFactory defaultFactory = AliyunClientFactories.from(Maps.newHashMap());
-    Assert.assertTrue(
-        "Should load default when factory impl not configured",
-        defaultFactory instanceof AliyunClientFactories.DefaultAliyunClientFactory);
-    Assert.assertNull(
-        "Should have no Aliyun properties set", defaultFactory.aliyunProperties().accessKeyId());
+    assertThat(defaultFactory instanceof AliyunClientFactories.DefaultAliyunClientFactory).as("Should load default when factory impl not configured").isTrue();
```
Comment (nastra) [inline aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java:35]: `assertThat(defaultFactory).as(..).isInstanceOf(AliyunClientFactories.DefaultAliyunClientFactory.class)`

File: aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java [hunk 2]
```diff
@@ -22,43 +22,30 @@
...
     AliyunClientFactory defaultFactoryWithConfig =
         AliyunClientFactories.from(ImmutableMap.of(AliyunProperties.CLIENT_ACCESS_KEY_ID, "key"));
-    Assert.assertTrue(
-        "Should load default when factory impl not configured",
-        defaultFactoryWithConfig instanceof AliyunClientFactories.DefaultAliyunClientFactory);
-    Assert.assertEquals(
-        "Should have access key set",
-        "key",
-        defaultFactoryWithConfig.aliyunProperties().accessKeyId());
+    assertThat(defaultFactoryWithConfig instanceof AliyunClientFactories.DefaultAliyunClientFactory).as("Should load default when factory impl not configured").isTrue();
```
Comment (nastra) [inline aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java:40]: same as above

File: aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java [hunk 3]
```diff
@@ -22,43 +22,30 @@
...
   }
 
   @Test
   public void testLoadCustom() {
     Map<String, String> properties = Maps.newHashMap();
     properties.put(AliyunProperties.CLIENT_FACTORY, CustomFactory.class.getName());
-    Assert.assertTrue(
-        "Should load custom class",
-        AliyunClientFactories.from(properties) instanceof CustomFactory);
+    assertThat(AliyunClientFactories.from(properties) instanceof CustomFactory).as("Should load custom class").isTrue();
```
Comment (nastra) [inline aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java:48]: same as above

**Review by nastra — COMMENTED** [review]
I think the `iceberg-aliyun` project is small enough so that we could include maybe all test classes and convert them to Junit5 as part of this PR

File: aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java (hunk 3 above)
Comment (DaVincii) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java:48]: @nastra Thanks for the input, migrating Aliyun is bit complex as it contains test rules which needs to migrated to registerClass and extendWith in junit 5. For now I will park this and try the simpler ones and once I gain some expertise will look into this.

Comment (zinking) [reply aliyun/src/test/java/org/apache/iceberg/aliyun/TestAliyunClientFactories.java:48]: I was thinking like 'is this PR made by the Text-Davinci openAI ?',  fortunately not.

**Comment by nastra** [comment]
@DaVincii are you planning to work on this? Would be great to move the aliyun project to JUnit5

**Comment by DaVincii** [comment]
> @DaVincii are you planning to work on this? Would be great to move the aliyun project to JUnit5

@nastra Sorry, got busy with some other work, I have made the review changes.  Are you asking for the migrating the classes which extend Rule class, as well?

**Comment by nastra** [comment]
I think at the very least we should switch the assertions to AssertJ in the `aliyun` module. Also I think we can convert `TestOSSURI` to JUnit5

**Pull request closed without merging**